bin/spring

```diff
@@ -1,10 +1,17 @@
 #!/usr/bin/env ruby
-if !defined?(Spring) && [nil, "development", "test"].include?(ENV["RAILS_ENV"])
-  # Load Spring without loading other gems in the Gemfile, for speed.
-  require "bundler"
-  Bundler.locked_gems.specs.find { |spec| spec.name == "spring" }&.tap do |spring|
+
+# This file loads Spring without using Bundler, in order to be fast.
+# It gets overwritten when you run the `spring binstub` command.
+
+unless defined?(Spring)
```

Comment:
So I think this is actually what breaks your deploy. The condition used to check that the environment is something that we have spring in - but it does not anymore. So it will attempt to load it, well, everywhere.